src/CoreBundle/Session/ChameleonSessionManager.php

```diff
@@ -174,6 +179,14 @@ public function boot()
         $request->getSession()->start();
         $this->isSessionStarting = false;
         $this->registerSessionInContainer($request->getSession());
+
+        if (null !== $memcache && null !== $memcache->getDriver()) {
+            $sessionInitResult = $memcache->getDriver()->getResultCode();
```

Comment:
I don't think, that this is the correct place to check if memcache is available. The user may well have choosen to use the database a session storage.

Instead, I would suggest moving the code into the boot method where the memcache connection is opened.

Reply:
This method here is called boot() (?)

Reply:
that is true :) but take a look at line 103

Reply:
If I remember correctly this location was chosen to be placed "after initialization - which is happening outside of the scope of this method".

It checks if memcache is actually used before doing a check (null !== $memcache).

Reply:
I am still not ok with this. for one, checking if memcache is null or not is not the same as checking if the user plans on using memcache for session storage (even if that may currently act the same).

I think we should solve this problem one of two ways:

We have the sanity check bundle and use it to check if the session memcache is available and works
Or we check if the where able to fetch/get data from memcache when accessing memcache - if the request returns an error that indicates, that the server has gone away, we can throw the exception - of course only for session access, not for cache access.

But doing the check here is not helpful. Either the configuration is invalid - which we should not check on every single call, or the memcache server has gone away - which we should check when accessing it.